packages/node/src/indexer/indexer.manager.ts

```diff
@@ -49,8 +50,13 @@ export class IndexerManager {
     });
     const tx = await this.sequelize.transaction();
     this.storeService.setTransaction(tx);
+
+    const inject = block.specVersion !== this.prevSpecVersion;
+    this.prevSpecVersion = block.specVersion;
+
     try {
-      await timeout(this.apiService.setBlockhash(block.block.hash), 10); //TODO remove this when polkadot/api issue #3197 solved
+      const hash = block.block.hash;
+      await timeout(this.apiService.setBlockhash(hash, inject), 10); //TODO remove this when polkadot/api issue #3197 solved
```

Comment:
btw, the timeout here can be removed, because 3.11.1 has fixed the issue.

Reply:
updated